test/unit/system_database_test.rb

```diff
@@ -0,0 +1,18 @@
+require 'test_helper'
+
+class SystemDatabaseTest < ActiveSupport::TestCase
+
+  def test_configuration_specification_is_memoized
+    FakeFS do
+      expected_conf = System::Database.configuration_specification
+      assert_instance_of ActiveRecord::ConnectionAdapters::ConnectionSpecification, expected_conf
+      config = Rails.root.join('config', 'database.yml')
+
+      FakeFS::FileSystem.clone(config.dirname.to_s)
+      FakeFS::FileUtils.rm(config.to_s)
+      conf = System::Database.configuration_specification
+
+      assert_same expected_conf, conf
```

Comment:
Can we assert how many times the method is called instead?

Reply:
Personally I prefer the way that it is tested, that even if the file is not there anymore, it works anyway. Because it is what will happen in real life through `admin/api/service_features#index` so it is a closer simulation that what we actually want to happen.